js/plot/util/fill.js

```diff
@@ -66,17 +68,7 @@ module.exports = function(pool, opts) {
         .attr({
           cursor: opts.cursor ? opts.cursor : 'auto',
           x: function(d, i) {
-            if (opts.dataGutter) {
-              if (i === currentData.length - 1) {
-                return fill.xPosition(d) - opts.dataGutter;
-              }
-              else {
-                return fill.xPosition(d);
-              }
-            }
-            else {
-              return fill.xPosition(d);
-            }
```

Comment:
Ack, I didn't mean delete the `opts.dataGutter` stuff everywhere, sorry! It is needed on weekly view. But for the midnight line where everything is inside a `if (opts.isDaily)` it's not necessary.